helper/resource/plan_check.go

```diff
@@ -0,0 +1,44 @@
+package resource
+
+import (
+	"github.com/hashicorp/go-multierror"
+	tfjson "github.com/hashicorp/terraform-json"
+	"github.com/mitchellh/go-testing-interface"
+)
+
+// TODO: document
+type PlanCheck interface {
+	RunCheck(req PlanCheckRequest, resp *PlanCheckResponse)
+}
+
+// TODO: document
+type PlanCheckRequest struct {
+	Plan *tfjson.Plan
+}
+
+// TODO: document
+type PlanCheckResponse struct {
+	Error    error
+	SkipTest bool
```

Comment:
May be worth considering making this a `string` so it can be immediately useful for the testing code to call `(*testing.T).Skip()` 👍 Hopefully some of this additionally suggested documentation here will provide a clue that skipping is a little bit more nuanced than immediately calling that method.

```suggestion
	// Skip, if non-empty, immediately skips further TestStep and defines a message
	// to be included with a call to (*testing.T).Skip(), which is visible in the test output.
	//
	// If a state has been applied via this TestStep or a prior TestStep, the testing will still
	// invoke Terraform to destroy that state before finalizing the skipped test result.
	Skip string
```

Reply:
By the way, skipping may turn out to be very complicated. If we want to save that for a followup implementation so we can at least get the error handling functionality out, totally cool. 😎

Reply:
So I was looking at this a little bit as I've never used `t.Skip` before, and as you noted it will halt execution of the test and run all `defer` functions, of which I found three:
- https://github.com/hashicorp/terraform-plugin-testing/blob/9c304ac8cd50c6dfd3c382237ff390f3297fc597/helper/resource/testing_new.go#L62
- https://github.com/hashicorp/terraform-plugin-testing/blob/9c304ac8cd50c6dfd3c382237ff390f3297fc597/helper/resource/testing_new.go#L388
- https://github.com/hashicorp/terraform-plugin-testing/blob/9c304ac8cd50c6dfd3c382237ff390f3297fc597/helper/resource/testing.go#L783

All of those will be run properly, but the question could also be, is it desirable for a skip in a `TestStep` to skip the entire `TestCase`?

Are there other considerations you had in mind?

Reply:
Hmm... maybe it's best to put the feature off to gather more use-cases. Maybe when we start looking at other skipping functionality like: https://github.com/hashicorp/terraform-plugin-testing/issues/68

I'll remove the Skip references from the RFC + this PR and create a new issue

Reply:
Created an issue: https://github.com/hashicorp/terraform-plugin-testing/issues/108